File: src/main/lib/trpc/project-config.ts
import { z } from 'zod'
import { router, publicProcedure } from './trpc'
import {
  getProjectConfig,
  hasRunCommand
} from '../project-config'

export const projectConfigRouter = router({
  get: publicProcedure
    .input(z.object({ path: z.string() }))
    .query(({ input }) => {
      return getProjectConfig(input.path)
    }),

  hasRunCommand: publicProcedure
    .input(z.object({ path: z.string() }))
    .query(({ input }) => {
      return hasRunCommand(input.path)
    }),

  runCommand: publicProcedure
    .input(z.object({ path: z.string() }))
    .mutation(({ input }) => {
      const config = getProjectConfig(input.path)
      if (!config.runCommand) {
        throw new Error('No run command configured for this project')
      }
      return { command: config.runCommand }
    })
})

File: src-tauri/src/commands/project_config.rs
use crate::project_config;

#[tauri::command]
pub fn project_config_get(path: String) -> Result<project_config::ProjectConfig, String> {
    Ok(project_config::get_project_config(&path))
}

#[tauri::command]
pub fn project_config_has_run_command(path: String) -> Result<bool, String> {
    Ok(project_config::has_run_command(&path))
}

#[derive(serde::Serialize)]
pub struct RunCommandResult {
    pub command: String,
}

#[tauri::command]
pub fn project_config_run_command(path: String) -> Result<RunCommandResult, String> {
    let config = project_config::get_project_config(&path);
    match config.run_command {
        Some(cmd) if !cmd.is_empty() => Ok(RunCommandResult { command: cmd }),
        _ => Err("No run command configured for this project".to_string()),
    }
}
